### PR TITLE
Fix CRTLIB error checking when connecting + other error checkings

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
Fixes https://github.com/codefori/vscode-ibmi/issues/1739

This PR adds a method that will parse commands `stderr` into `IBMiMessages` to facilitate error analysis.
This helps checking what IBM i messages are returned after a command call and fixes some checking that only looked for the first returned message ID.

Also fixes a very weird bug that the core team have only been able to recreate on pub400:

```
qsh: 001-0019 Error found searching for command [[. No such path or directory.
```

This new test would fail without the one line change I made in CompileTools#runCommand.

### Checklist
* [x] have tested my change

### CODE REVIEW

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
@sebjulliand When connecting to PUB400, I get the following error:

image

I checked, and there are no dataarea like that on PUB400...?

**Comment by sebjulliand** [comment]
> @sebjulliand When connecting to PUB400, I get the following error:
> 
> I checked, and there are no dataarea like that on PUB400...?

@chrjorgensen The change I made doesn't impact this check. But it would mean that this call to `CHKOBJ` didn't return 0:
image

Do you have any more detail in debug, like the content of the `QCPTOIMPF` CommandResult?

**Comment by chrjorgensen** [comment]
@sebjulliand It's the previously discussed error with `[[`:

```
"CPF9801:  Object QCPTOIMPF in library QSYS not found.\nqsh: 001-0019 Error found searching for command [[. No such path or directory."
```

**Comment by chrjorgensen** [comment]
@sebjulliand This is the status after the `CRTLIB` command:

image

Should additional tests for message ID's be added? CPF0001?

**Comment by worksofliam** [comment]
@chrjorgensen 

> I actually don't understand how the error code can be zero, when the command failed...!?

I don't think it can. This is likely my bad for sure!

**Comment by worksofliam** [comment]
@chrjorgensen that is totally weird, but I'm guessing it's because of how runCommand returns the exit code. Defo got something to do with those brackets.

This line of code: https://github.com/codefori/vscode-ibmi/blob/7c7dfb42d6cbf7cf7cb43c36072a206ed32dbf33/src/api/CompileTools.ts#L663

**Comment by worksofliam** [comment]
I have merged my PR into this one to make it easier. Now I am no longer getting the warning about the data area existing! 

@chrjorgensen back to you for some testing and I am going to do a quick code review too.

**Review by sebjulliand — COMMENTED** [review]
Works fine here on PUB400 and other systems.
Looks like we could merge!

**Comment by worksofliam** [comment]
@sebjulliand I agree on the merge, but I am thinking we should let @chrjorgensen have another go through as well.

File: src/api/IBMi.ts [hunk 1]
```diff
@@ -348,47 +348,30 @@ export default class IBMi {
           command: `CRTLIB LIB(` + this.config.tempLibrary + `) TEXT('Code for i temporary objects. May be cleared.')`,
           noLibList: true
         });
+        console.log(createdTempLib?.stderr);
```
Comment (worksofliam) [inline src/api/IBMi.ts:351]: Should this be taken out?

Comment (sebjulliand) [reply src/api/IBMi.ts:351]: Agreed

File: src/api/CompileTools.ts [hunk 2]
```diff
@@ -660,7 +660,7 @@ export namespace CompileTools {
               command: [
                 ...options.noLibList? [] : buildLiblistCommands(connection, ileSetup),
                 ...commands.map(command =>
-                  `${`system ${GlobalConfiguration.get(`logCompileOutput`) ? `` : `-s`} "${command.replace(/[$]/g, `\\$&`)}"; exit $?`}`,
+                  `${`system ${GlobalConfiguration.get(`logCompileOutput`) ? `` : `-s`} "${command.replace(/[$]/g, `\\$&`)}" && exit $?`}`,
```
Comment (chrjorgensen) [inline src/api/CompileTools.ts:663]: @worksofliam How can this work? AFAIK, `&&` make the `exit` only being run if the previous command runs successfully?

Comment (worksofliam) [reply src/api/CompileTools.ts:663]: You are completely right. I am taking out the `exit` in a new commit and the tests still pass.

**Review by chrjorgensen — APPROVED** [review]
@sebjulliand @worksofliam Working as expected - approved!

**Pull request merged**